pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -2216,4 +2219,24 @@ public int getBrokerDeleteInactiveTopicsMaxInactiveDurationSeconds() {
             return brokerDeleteInactiveTopicsMaxInactiveDurationSeconds;
         }
     }
+
+    public String getAppliedAdvertisedAddress() {
```

Comment:
CN we add a javadoc that explains what this method does and that this is an internal method not intended to be used as public API?

Reply:
OK, I will add the document.
But this method needs to be exposed, not an internal method.
I named it getAdvertisedAddress before, but this will cause problems because AdvertisedAddress and AdvertisedListener cannot be set at the same time.

Reply:
I am fine with making this method "public", we must do it.
But I would like to suggest to the users that this method is a convenience method, used by Pulsar Internals.

in fact this is not a plain property with getter/setter but it is computed.

if we add this method to this class, that is part of the Public API, then users will be able to rely on it and if we want to change it we will have to take into account backward compatibility.

I known that I am on the edge of nitpicking, but if we add a method to a public, widely used API, we will have to maintain it

Reply:
that would be great !

Reply:
This tool class will not be exposed to users. Only exists in Broker.